### DISCUSSION TITLE
How do I save Slick Grid data when Add New Row ?

### DISCUSSION THREAD

**Discussion by eissa1005** [serenity-is/Serenity#5832] [Help] [answered]
I changed Options Grid  so  that a new row can be added => ( Inline Or  Batch) 
But How do I save Slick Grid data When Save Master 

![Grid1](https://user-images.githubusercontent.com/20481378/123553542-0aada200-d77c-11eb-858e-df3970ee9f30.png)
![Grid2](https://user-images.githubusercontent.com/20481378/123553546-0d0ffc00-d77c-11eb-88bb-424b0177458f.png)

**Comment by VictorTomaili** [comment]
You need to get items from your grid editor then send this data to an endpoint that accepts this bulk data when clicked to save button. 
Use `this.getItems()` forget all items in grid. Enure about these items must be updated after any data changed by inline.

**Comment by eissa1005** [comment]
@VictorTomaili 
Note when adding a newRow , I did not use Dialog ( Edit Modes) but use Method SetValue to Grid (Do not use Memory Grid)
When Click Save All in Dialog Master occur error  Could not find field "idProperty"
![Untitled](https://user-images.githubusercontent.com/20481378/123835706-2a260580-d909-11eb-9a90-e5f8564b2b2a.png)

**Reply by VictorTomaili** [reply]
For serenity, we didn't have `inline grid add functionality` by default. But anyway you can create a dummy unique id for the new rows that will be ignored on the server-side while inserting.

**Reply by eissa1005** [reply]
@VictorTomaili 
 any example 
Another question please How do I connect  Slick.Data.DataView  with Slick.Grid  
When call Event OnRowsChanged Not Fire

**Comment by eissa1005** [comment, marked as answer]
Resolve Done // Add => Update => Delete

**Reply by ra-bedo11** [reply]
How Solved ?
